### ISSUE TITLE
Faster way to package external CSS/JS files

### ISSUE THREAD

**Issue by MachHach** [Raphx/vmlauncher-client#5]
Current implementation of packaging external CSS/JS files via Webpack has significant increase in processing time (as compared to what was used previously).
While using `npm run webpack-dev` to hot-reload the application, some times Webpack can get stuck on errors and stop hot-reloading, requiring a restart of the hot-reloader. Each restart will incur the above mentioned processing duration, and are cumulatively expensive when active development is in progress.

Open for discussion.

**Comment by Raphx** [comment]
I'm starting to see that it may be a good idea to bundle CSS file separately from the bundled JS. This allows the browser (Chromium in Electron) to load CSS and JS files in parallel.

  1. For CSS files, it may be good to use [this](https://github.com/webpack-contrib/extract-text-webpack-plugin) alongside with `css-loader` to separately bundle CSS

  2. For JS files, e.g. jQuery and Materialize static JS, still investigating for options

  3. For application JS files, we will be bundling these the way we are doing now (and perhaps combine this with (2)? )

Not sure yet if this separation in bundling will improve performance. I will experiment around on a branch soon.

**Comment by Raphx** [comment]
See #8 if it fits our use case. We can further improve on it though.

If it is good, you can rebase and merge into `master`. 😄

**Comment by MachHach** [comment]
The performance of the development webpack `npm run watch` does not differ by much from before, it bundles a massive 550 hidden modules as compared to 579 hidden modules by `npm run build`.

Suspecting the use of _material-ui_ package is one of the reasons there.

**Comment by Raphx** [comment]
Since `webpack` bundles the node module dependencies, the more dependencies we use, the longer it takes to bundle.

There's an extent to which we can optimize in `webpack` itself 😖 

Further optimization can be done on the programmatic aspect, i.e. by reducing the number of dependencies for the project.

The `walker` module in this project uses the `lodash` package for one simple task. I will look into this one in accordance to our desired performance goal.

**Comment by Raphx** [comment]
Project maintenance ended.
